scripts/release.js

```diff
@@ -0,0 +1,189 @@
+    `${chalk.yellow('⁉︎')} Proceed anyway? [y/n]`,
+    promptOpts
+  );
+  if (answer === 'n') {
+    logWarning(`Stopping right here.`);
+    process.exit(0);
+  }
+};
+
+const dryRunPrompt = async () => {
+  let answer = await prompt(
+    `${chalk.blue('⁇')} This is ${chalk.yellow('NOT')} a --dry-run.` +
+      leftPad1(`Did you complete a successful --dry-run first? [y/n]`),
+    promptOpts
+  );
+  if (answer === 'n') await proceedAnywayPrompt();
+};
+
+(async () => {
+  try {
+    if (!dryRun) await dryRunPrompt();
+
+    logInfo(`Determining the current branch...`);
+
+    let currentBranch;
+    try {
+      currentBranch = execSync(`git rev-parse --abbrev-ref HEAD`)
+        .toString()
+        .trim();
+    } catch (e) {
+      logError(`Couldn't determine the branch. Please check the error above.`);
+      throw new Error();
+    }
+
+    if (currentBranch !== branch) {
+      logError(
+        `Current branch '${currentBranch}' is not the same as release branch`,
+        `'${branch}'. Please checkout the release branch before rerunning this`,
+        `script or rerun with '--repo-branch ${currentBranch}'.`
+      );
+      throw new Error();
+    }
+
+    logSuccess(`Current branch and relase branch are the same.`);
+    logInfo(`Checking the working tree...`);
+
+    try {
+      execSyncInherit(`npm run --silent check-working-tree`);
+      logSuccess(`Working tree is clean.`);
+    } catch (e) {
+      logError(
+        `Working tree is dirty or has untracked files. Please make necessary`,
+        `changes or commits before rerunning this script.`
+      );
+      throw new Error();
+    }
+
+    logInfo(
+      `Fetching from origin '${origin}' to compare local and remote branches...`
+    );
+
+    try {
+      execSyncInherit(`git fetch ${origin}`);
+    } catch (e) {
+      logError(`Couldn't fetch latest commits. Please check the error above.`);
+      throw new Error();
+    }
+
+    let localRef, originRef;
+    try {
+      localRef = execSync(`git rev-parse ${branch}`).toString();
+      originRef = execSync(`git rev-parse ${origin}/${branch}`).toString();
+    } catch (e) {
+      logError(`A problem occured. Please check the error above.`);
+      throw new Error();
+    }
+
+    if (localRef !== originRef) {
+      logError(
+        `Local branch '${branch}' is not in sync with '${origin}/${branch}'.`,
+        `Please sync branches before rerunning this script.`
+      );
+      throw new Error();
+    }
+
+    logSuccess(`Local branch is in sync with remote branch.`);
+    logInfo(`Running Standard Version${dryRunMaybe()}...`);
+
+    await standardVersion({
+      dryRun,
+      prerelease,
+      releaseAs,
+      sign
+    });
+
+    logInfo(`Publishing new Embark version on npm${dryRunMaybe()}...`);
+
+    const npmPublishCommand = [
+      `npm publish`,
+      leftPad1Maybe(`${distTag ? `--tag ${distTag}` : ''}`),
+      leftPad1Maybe(`${dryRun ? '--dry-run' : ''}`)
+    ].join('');
```

Comment:
Don't we wanna join with a space here? Otherwise we'd end up with `npm publish--tag foo--dry-run` no?

Reply:
Ah nevermind, there's a `leftPad1Maybe`. Still, wouldn't it be simpler to remove that and just join with `' '` instead?

Reply:
We could, but it would lead to extra spaces in the command string when `distTag` or `dryRun` is empty — not the end of the world but I'd prefer to not have extraneous spaces.

The idea with `leftPad1/Maybe` was to avoid a situation where someone in the future working on the script removes a space from a string because it looks like it was a typo and then, because testing the script is mostly a manual process, the mistake isn't realized until later. A function call is less likely to be accidentally removed, I think.